flink-yarn/src/test/java/org/apache/flink/yarn/AbstractYarnClusterTest.java

```diff
@@ -82,26 +83,21 @@ private ApplicationReport createApplicationReport(
 		ApplicationId applicationId,
 		YarnApplicationState yarnApplicationState,
 		FinalApplicationStatus finalApplicationStatus) {
-		return ApplicationReport.newInstance(
-			applicationId,
-			ApplicationAttemptId.newInstance(applicationId, 0),
-			"user",
-			"queue",
-			"name",
-			"localhost",
-			42,
-			null,
-			yarnApplicationState,
-			null,
-			null,
-			1L,
-			2L,
-			finalApplicationStatus,
-			null,
-			null,
-			1.0f,
-			null,
-			null);
+
+		ApplicationReport applicationReport = Records.newRecord(ApplicationReport.class);
```

Comment:
Seems `Records.newRecord(...)` is still unstable. Is there a stable approach to create `ApplicationReport`?

Reply:
Not that I'm aware of.
I admit that this is not a perfect solution and is not guaranteed compatible with all future hadoop versions. This approach only works with the current known hadoop versions.

Reply:
Ok. Let's take this workaround for now.